firebase-perf/src/main/java/com/google/firebase/perf/metrics/AppStartTrace.java

```diff
@@ -191,51 +215,57 @@ public synchronized void onActivityResumed(Activity activity) {
     appStartActivity = new WeakReference<Activity>(activity);
 
     onResumeTime = clock.getTime();
-    final Timer startTime = FirebasePerfProvider.getAppStartTime();
+    this.appStartTime = FirebasePerfProvider.getAppStartTime();
+    this.startSession = SessionManager.getInstance().perfSession();
     AndroidLogger.getInstance()
         .debug(
             "onResume(): "
                 + activity.getClass().getName()
                 + ": "
-                + startTime.getDurationMicros(onResumeTime)
+                + this.appStartTime.getDurationMicros(onResumeTime)
                 + " microseconds");
 
+    // Log the app start trace in a non-main thread.
+    executorService.execute(this::logAppStartTrace);
+
+    if (isRegisteredForLifecycleCallbacks) {
+      // After AppStart trace is logged, we can unregister this callback.
+      unregisterActivityLifecycleCallbacks();
```

Comment:
Just out of curiousity - Can this happen on non-main thread? Just thinking if we can delay this for later on a non-main thread.

Reply:
We cannot. If we delay this, it is possible that the app enters onActivityResumed multiple times before the executor executes so that we might log the startup time multiple times.